src/JustSaying/Naming/DefaultNamingConventions.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace JustSaying.Naming
+{
+    public class DefaultNamingConventions : ITopicNamingConvention, IQueueNamingConvention
+    {
+        private const int MaxTopicNameLength = 256;
+        private const int MaxQueueNameLength = 80;
+
+        private static readonly HashSet<Type> TypesToMapAutomatically = new HashSet<Type>
+        {
+            typeof(string),
+            typeof(object),
+            typeof(bool),
+            typeof(byte),
+            typeof(char),
+            typeof(decimal),
+            typeof(double),
+            typeof(short),
+            typeof(int),
+            typeof(long),
+            typeof(sbyte),
+            typeof(float),
+            typeof(ushort),
+            typeof(uint),
+            typeof(ulong),
+            typeof(void),
+            typeof(TimeSpan),
+            typeof(DateTime),
+            typeof(DateTimeOffset)
+        };
+
+        public string TopicName<T>() => CreateResourceName(typeof(T), MaxTopicNameLength);
```

Comment:
Maybe we could make these two methods `virtual` so you could create a derived class to override just one if you so desired, or if you wanted to override both to do something simple like add on a prefix or suffix?